October 2025 backports to 3.2

vadz opened the pull request:
Please suggest more commits to backport here.

vadz committed:
Fix fatal assert when using box indicators in wxStyledTextCtrl

This assert, generated from wxEVT_PAINT handler, happened because
SurfaceImpl class unconditionally casts its wxDC to wxMemoryDC before
creating wxGraphicsContext from it, but the actual wxDC was a wxPaintDC,
created in wxStyledTextCtrl::OnPaint(), and not a wxMemoryDC.

Fix this by creating wxBufferedPaintDC, which is a wxMemoryDC, instead.

Note that it is not clear how could this work before as OnPaint()
created wxPaintDC since always, and SurfaceImpl seems to have always
needed wxMemoryDC too. It would also be better to avoid casting wxDC to
wxMemoryDC in the first place, as this forces us to use wxMemoryDC even
on platforms using double-buffering, i.e. effectively unnecessarily
triple-buffering drawing here.

(cherry picked from commit e92f57b421938e00f0ad944c281c2e353eee2ab8)

See #25917.

Blake-Madden committed:
Clarify wxLog::SetActiveTarget()

See #25852.

(cherry picked from commit 5de9ceb603ab608bbcba563f584b854ac68bcaf8)

vslavik committed:
Fix nested markup conversion to NSAttributedString

wxMarkupToAttrStringBase implementation of markup parsing failed to
correctly apply nested attributes: in e.g. <i><b>foo</b></i>, only the
outer attribute was in effect.

This was due to applying - and overwriting - attributes to a span in
OnAttrEnd(), which is called first for the inner span, and then for the
outer one, overwriting all inner changes.

Instead, apply the currently effective attributes to text in OnText(),
similarly to how wxMarkupParserRenderOutput does it. As a side effect,
this also eliminates the need to apply default font to the entire range
first.

See #25864.

(cherry picked from commit ca05aee2f8134dcae53aecd1a952d2a57f51b223)

perazz committed:
Avoid using already destroyed timers in wxOSX

Check that the timer is still valid before using the associated object.

This is necessary because already queued timer events are still
delivered, even if wxTimer is destroyed.

See #25871.

(cherry picked from commit f25aec1b38bf208a34e0aabefae5e05ca265ecd3)

Blake-Madden committed:
Fix harmless unused parameter warning in wxOSX

Add missing WXUNUSED().

See #25880.

(cherry picked from commit 33b289bb9ba39c3ce207d172fdf3a6e33fdec16f)

Blake-Madden committed:
Check that wxRibbonBar tab icon is valid before using it

Using invalid item resulted in an assert, avoid it by checking if it is
valid before using it.

Also avoid calling GetIcon() and GetLogicalWidth() twice in a row by
storing the result in a variable.

See #25881.

(cherry picked from commit 3f9d34b4a161cb1aeeba04017691a48bb8889d60)

Co-authored-by: Vadim Zeitlin <vadim@wxwidgets.org>

vslavik committed:
Fix compilation with wxUSE_OWNER_DRAWN==0

Add missing wxUSE_OWNER_DRAWN check to wxMenuItem code.

Closes #25883.

(cherry picked from commit 7d91e361eb6fae730434df8f6f3abc4c17d91650)

vslavik committed:
Improve padding when using HiDPI in generic wxDataViewCtrl

It looked almost non-existent at 200%, so scale it with DPI to improve
the control appearance.

Closes #25882.

(cherry picked from commit 450aa484afd242d1ba9e43fe0d7a00b0b3b273af)

DietmarSchwertberger committed:
Fix selecting individual columns with Ctrl-Click in wxGrid

This didn't work correctly because the check if the column was already
selected in wxGridColumnOperations::IsLineInSelection() used incorrect
parameter order when calling IsInSelection().

See #25884.

(cherry picked from commit c7e5734797078dd293cf05ba683bcd7603eebf53)

Blake-Madden committed:
Fix bitmap position in wxDC::DrawLabel() in high DPI

Use bitmap size in logical, not physical, pixels to position it
correctly.

See #25888.

(cherry picked from commit 23083720a9f8d134c6d7b481d567e133d9384621)

vadz committed:
Revert to using FreeBSD 14.2 for Cirrus CI builds

FreeBSD 15 seems to support inotify() too now, but our tests still don't
pass when using it, probably because it is still implemented in terms of
kqueue() internally. Until this can be fixed, use the same FreeBSD 14.2
as in master.

csomor committed:
Fix handling filters in wxFileDialog in wxOSX

As allowedTypes were restricted to the first filter expression, choosing
another filter only showed the files conforming to both expressions
instead of only applying the selected filter.

This commit is best viewed ignoring whitespace-only changes.

Fixes wxWidgets/Phoenix/issues#2806

See #25918, #25922.

(cherry picked from commit b153aafee9d74ecfaef7b1447b79012e6e60c11a)

vadz committed:
Recalculate wxGenericTreeCtrl line height after font change

This ensures that the items always consume as much space as they need
vertically and not more.

See #16428.

(cherry picked from commit 3e937bbd19aaa500fdfd9593cd9cf79d0ef36d15)

swt2c commented:
@vadz, please include https://github.com/wxWidgets/wxWidgets/commit/b153aafee9d74ecfaef7b1447b79012e6e60c11a

vadz committed:
Fix compilation problem due to memset_s() when using PCH

Define __STDC_WANT_LIB_EXT1__ in wx/platform.h, which is included before
the standard headers even when using PCH, to make sure memset_s()
declaration is available in this case too.

See #24687.

(cherry picked from commit e913e9eba4815d5b2fe8148ed9b8b61e0441160f)
